packages/coin-support-tron/src/operations/syncAccount/index.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  createSyncAccountsObservable,
+  IGetAddressDetails,
+} from '@cypherock/coin-support-utils';
+import {
+  IAccount,
+  ITransaction,
+  TransactionStatusMap,
+  TransactionTypeMap,
+} from '@cypherock/db-interfaces';
+import { getAccountsTransactionsByAddress } from '../../services';
+
+import { BigNumber } from '@cypherock/cysync-utils';
+import lodash from 'lodash';
+import { ISyncTronAccountsParams } from './types';
+import { TronTransaction } from '../../services/validators';
+
+const PER_PAGE_TXN_LIMIT = 100;
+
+interface GetTransactionParserParams {
+  address: string;
+  account: IAccount;
+}
+
+const getTransactionParser = (
+  _params: GetTransactionParserParams,
+): ((transaction: any) => ITransaction) => {
+  const myAddress = _params.address;
+  const { account } = _params;
+
+  return (transaction: TronTransaction): ITransaction => {
+    const fromAddr = transaction.fromAddress;
+    const toAddr = transaction.toAddress;
+    const amount = transaction.value;
+    const fees = new BigNumber(transaction.fees).toFixed();
+    const timestamp = new BigNumber(transaction.blockTime)
+      .dividedBy(1_000_000)
+      .toNumber();
+
+    const txn: ITransaction = {
+      hash: transaction.txid,
+      accountId: account.__id ?? '',
+      walletId: account.walletId,
+      assetId: account.parentAssetId,
+      parentAssetId: account.parentAssetId,
+      familyId: account.familyId,
+      amount,
+      fees,
+      confirmations: 1,
+      status:
+        transaction.tronTXReceipt.status === 1
+          ? TransactionStatusMap.success
+          : TransactionStatusMap.failed,
+      type:
+        myAddress === fromAddr
+          ? TransactionTypeMap.send
+          : TransactionTypeMap.receive,
+      timestamp,
+      blockHeight: transaction.blockHeight,
+      inputs: [
+        {
+          address: fromAddr,
+          amount,
+          isMine: myAddress === fromAddr,
+        },
+      ],
+      outputs: [
+        {
+          address: toAddr,
+          amount,
+          isMine: myAddress === toAddr,
```

Comment:
Ignore case while comparing

Reply:
converted addresses to lowercase during declaration